Port to 1.21.5 and rework no vote data.

ChrysanthCow opened the pull request:
Apologies if any file spacing is incorrect due to my IDE settings.

This PR:
- Ports the mod to 1.21.4.
- Reworks having no voting data.
  - Removed ModFest 1.21 test data from being built into the mod. 
  - The game will no longer throw an exception upon not detecting categories or vote data. It will now have unique inactive voting handling if either one is not present.

ChrysanthCow committed:
Port to 1.21.4.

ChrysanthCow committed:
Bump version.

ChrysanthCow committed:
Allow and handle no voting data.

ChrysanthCow committed:
Fix missing .

ChrysanthCow committed:
Fix server-sided availability check.

acikek commented on src/main/java/net/modfest/ballotbox/client/BallotBoxClient.java:
```diff
@@ -14,12 +14,17 @@
 		return !client.isIntegratedServerRunning() && ClientPlayNetworking.canSend(OpenVoteScreen.ID);
 	}
 
+    public static boolean isAvailable() {
+        return available;
+    }
```
`available` is already public static, no need for accessor

ChrysanthCow replied:
Resolved as of https://github.com/ModFest/ballotbox/pull/4/commits/16e931ae97f52606a96793641ce1a85aed38a05d

acikek commented on src/main/java/net/modfest/ballotbox/client/BallotBoxClient.java:
```diff
@@ -14,12 +14,17 @@
 public class BallotBoxClient implements ClientModInitializer {
 	public static final Logger LOGGER = LoggerFactory.getLogger("%s-client".formatted(BallotBox.ID));
 	public static Instant closingTime = null;
+    public static boolean available = true;
```
I suggest renaming `available` to something more descriptive, e.g. `hasVotingOptions`

ChrysanthCow replied:
Resolved as of https://github.com/ModFest/ballotbox/pull/4/commits/16e931ae97f52606a96793641ce1a85aed38a05d

acikek commented on gradle.properties:
```diff
@@ -17,9 +17,9 @@ authors=ModFest
-branch=1.21
-tagBranch=1.21
-compatibleVersions=1.21, 1.21.1
-compatibleLoaders=fabric, quilt, neoforge
```
Why remove neoforge?

ChrysanthCow replied:
Ah, didn't see this one. The reason I removed it is because Connector isn't on this version and probably won't be.

sleepingdragoninn replied:
yep, that's fine

acikek commented:
Did you test ballotbox data via a data pack?

ChrysanthCow commented:
> Did you test ballotbox data via a data pack?

I did!

ChrysanthCow commented:
![image](https://github.com/user-attachments/assets/ca5d9929-e974-4055-b872-8e25bd315417)

ChrysanthCow commented:
Currently testing icons.

ChrysanthCow committed:
Rename `available` to `hasVotingOptions`.

ChrysanthCow commented:
![image](https://github.com/user-attachments/assets/00f59902-076d-4cff-bef2-b6f3913ad76b)

ChrysanthCow committed:
Fix mod icons in the voting screen.

ChrysanthCow commented:
Hi, minor update for this PR.

We don't know whether Mod Garden will be running on 1.21.4 or 1.21.5, simply because we don't know when .5 will release. 

If it releases in like March and if there's enough time to update mods, this PR will be changed to .5, so don't merge until we find out which version we're running on.

sleepingdragoninn commented:
I kinda want everything in this PR except the 1.21.4 port - any chance of that?

ChrysanthCow commented:
> I kinda want everything in this PR except the 1.21.4 port - any chance of that?

Sure! I'll get to it sometime soon!

sylv256 committed:
Merge remote-tracking branch 'origin/1.21' into 1.21.5

# Conflicts:
#	gradle.properties
#	libs.versions.toml
#	src/main/java/net/modfest/ballotbox/BallotBoxPlatformClient.java
#	src/main/java/net/modfest/ballotbox/client/VotingScreen.java
#	src/main/java/net/modfest/ballotbox/mixin/client/GameMenuScreenMixin.java
#	src/main/java/net/modfest/ballotbox/mixin/client/TitleScreenMixin.java

sylv256 committed:
build(gradle): add simple resource loader

sylv256 committed:
build(gradle): test source-set with testing resources

sylv256 committed:
fix: Mod Menu "Mods" button overlapping Report Bugs replacements

sylv256 committed:
fix: sensible defaults

sylv256 committed:
chore: add contributors (Calico, Sylv)

I don't know if this is proper, but considering the port I thought it would be correct to add us as contributors.

sylv256 committed:
chore: Update to 1.21.5

sylv256 committed:
fix(NBT/PState): 1.21.5 porting bugs

sylv256 commented:
1.21.5 port complete :pineapple: 
![image](https://github.com/user-attachments/assets/1de0efd9-ef04-4f9e-953f-6bac6d07f9ac)
